testing/test_config.py

```diff
@@ -847,9 +869,9 @@ def test_inifilename(self, tmpdir):
                 """
             )
         )
-        with cwd.ensure(dir=True).as_cwd():
-            config = Config.fromdictargs(option_dict, ())
-            inipath = py.path.local(inifile)
+        os.chdir(cwd)
+        config = Config.fromdictargs(option_dict, ())
+        inipath = py.path.local(inifile)
```

Comment:
I believe the problem lies here -- the previous code ensured that the dir is changed back after the `with` scope and this seems needed.

I suggest writing this like this:

```suggestion
        with MonkeyPatch.context() as mp:
            mp.chdir(cwd)
            config = Config.fromdictargs(option_dict, ())
            inipath = py.path.local(inifile)
```

Reply:
Good catch! This makes more sense now - updated in 009e04c